src/asm/rpn.c

```diff
@@ -478,6 +514,16 @@ void rpn_BinaryOp(enum RPNCommand op, struct Expression *expr,
 		expr->val = sym_GetValue(symbol1) - sym_GetValue(symbol2);
 		expr->isKnown = true;
 	} else {
+		if (op == RPN_AND) {
```

Comment:
IMO this should be written more like the analogous case above, `op == RPN_SUB && isDiffConstant(src1, src2)`. So:

```c
} else if (op == RPN_AND && (constMaskVal = tryConstMask(src1, src2)) != -1) {
    expr->val = constMaskVal;
    expr->isKnown = true;
} else {
    /* If it's not known, start computing the RPN expression */
    ...
}
```

(`int32_t constMaskVal;` would have to be declared before the whole `if` block; that could be avoided in C++ with an initializer statement, like `if (int32_t constMaskVal = tryConstMask(src1, src2); op == RPN_ADD && constMaskVal != -1)`.)